workers/data_refinery_workers/downloaders/utils.py

```diff
@@ -65,7 +65,18 @@ def create_processor_jobs_for_original_files(original_files: List[OriginalFile],
     """
     for original_file in original_files:
         sample_object = original_file.samples.first()
-        pipeline_to_apply = determine_processor_pipeline(sample_object)
+
+        if original_file.filename[-4:] == ".xml":
+            logger.info("Skipping useless file",
+                file=original_file.id,
+                filename=original_file.filename)
+            continue
+
+        # We NO_OP processed data. It's what we do.
+        if '.processed' in original_file.source_url:
+            pipeline_to_apply = ProcessorPipeline.NO_OP
+        else:
+            pipeline_to_apply = determine_processor_pipeline(sample_object)
```

Comment:
I think this logic should be in `determine_processor_pipeline` so that anywhere else that calls this also has this check (like the other call to this function in `create_processor_job_for_original_files`).

Reply:
The difference is that `determine_processor_pipeline` is entirely Sample based, and can only return a valid pipeline. This is a check based on a file, and particularly a file which will _never_ have a valid associated pipeline. I see this more as a preprocessor check. In `create_processor_job_for_original_files`, we always want to include all of the supplied files. There isn't anywhere else where `determine_processor_pipeline` is used.